compiler/rustc_driver/src/lib.rs

```diff
@@ -581,18 +586,18 @@ pub fn try_process_rlink(sess: &Session, compiler: &interface::Compiler) -> Comp
             sess.init_crate_types(collect_crate_types(sess, &[]));
             let outputs = compiler.build_output_filenames(sess, &[]);
             let rlink_data = fs::read(file).unwrap_or_else(|err| {
-                sess.fatal(&format!("failed to read rlink file: {}", err));
+                sess.emit_fatal(RlinkUnableToRead { err });
             });
             let codegen_results = match CodegenResults::deserialize_rlink(rlink_data) {
```

Comment:
Do you think you could modify `deserialize_rlink` to return an enum with the different error messages so that we can translate them here?

Reply:
Yeah, I can surely have a try at that!

Reply:
I wasn't expecting that the enum returned by `deserialize_rlink` would itself derive `SessionDiagnostic` or `SessionSubdiagnostic`, but that we'd have some matching structs in `rustc_driver` for each of the error kinds and we'd use those. If we add `SessionDiagnostic` for enums later then we can tidy this up, but for now, just have `deserialize_rlink`'s error be a plain ol' enum.

Reply:
Haha I knew I went too far with this stupid idea of mine 🙈